### PR TITLE
dev envs refresh

### PR DESCRIPTION
**Pull request by aevesdocker**

### CODE REVIEW

**Comment by hyu** [comment]
This looks great! I noticed these changes:


- Overview has "How Does it Work" section, with Pre-reqs moved to the appropriate tutorial pages.
  - "Dev Environments is powered by [Docker Compose](https://deploy-preview-15989--docsdocker.netlify.app/compose/)." is a great way of putting it!
  - Nice shoutout to `docker dev`
- Instead of Simple/Advanced, use the more functional Launch/Set up framing.
- New section for `docker dev` details

I agree these changes make the docs more helpful and orienting!

**Review by hyu — APPROVED** [review]
This looks great as is! I added a suggestion, but it's not urgent.

File: desktop/dev-environments/index.md [hunk 1]
```diff
@@ -11,48 +11,24 @@ Dev Environments lets you create a configurable developer environment with all t
...
+{: .important}
 
-The following section lists known issues and workarounds:
+Dev Environments is powered by [Docker Compose](/compose/). This allows Dev Environments to take advantage of all the benefits and features of Compose whilst adding an intuitive GUI where you can launch environments with the click of a button.
 
-1. When sharing a dev environment between Mac and Windows, the VS Code terminal may not function correctly in some cases. To work around this issue, use the Exec in CLI option in the Docker Dashboard.
+Every dev environment you want to run needs a `compose-dev.yaml` file which configures your application's services and lives in your project directory. You don't need to be an expert in Docker Compose or write a `dev-compose.yaml` file from scratch - there are many [sample dev environments](https://github.com/docker/awesome-compose) that provide a useful starting point for how to integrate different services. Alternatively, see [Set up a dev environment](set-up.md) for more information.
```
[the remark this answers is not in the document]
Comment (aevesdocker) [reply desktop/dev-environments/index.md]: ✔️

File: desktop/dev-environments/set-up.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+RUN apk update \
+    && apk add git
+CMD ["go", "run", "main.go"]
+FROM alpine:3.12
+EXPOSE 8000
+COPY --from=build /go/src/github.com/org/repo/server /server
+CMD ["/server"]
+```
+
+The `development`target uses a `golang:1.16-alpine` image with all dependencies you need for development. You can start your project directly from VS Code and interact with the others applications or services such as the database or the frontend.
```
Comment (dvdksn) [inline desktop/dev-environments/set-up.md:82]:
```suggestion
The `development` target uses a `golang:1.16-alpine` image with all dependencies you need for development. You can start your project directly from VS Code and interact with the others applications or services such as the database or the frontend.
```

**Review by dvdksn — COMMENTED** [review]
only spotted a tiny whitespace issue, otherwise lgtm :))

**Review by dvdksn — APPROVED** [review]

**Pull request merged**